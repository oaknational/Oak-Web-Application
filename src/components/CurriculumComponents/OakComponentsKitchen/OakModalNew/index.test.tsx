import { act } from "@testing-library/react";

import { OakModalNew } from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

window.matchMedia = jest.fn().mockReturnValue({
  matches: true,
});

describe("OakModalNew", () => {
  it("should match snapshot", () => {
    HTMLDialogElement.prototype.close = () => {};
    HTMLDialogElement.prototype.showModal = () => {};
    const { container, getByTestId } = renderWithTheme(
      <OakModalNew
        title={"test_title"}
        content={"test_content"}
        footer={"test_footer"}
        open={false}
        onClose={() => {}}
      />,
    );

    expect(getByTestId("modal").tagName).toEqual("DIALOG");
    expect(getByTestId("modal-title")).toHaveTextContent("test_title");
    expect(getByTestId("modal-content")).toHaveTextContent("test_content");
    expect(getByTestId("modal-footer")).toHaveTextContent("test_footer");
    expect(container).toMatchSnapshot();
  });

  it("should open/close when open state changed", () => {
    const closeMock = jest.fn();
    const showModalMock = jest.fn();
    const onCloseMock = jest.fn();
    HTMLDialogElement.prototype.close = closeMock;
    HTMLDialogElement.prototype.showModal = showModalMock;
    const { rerender } = renderWithTheme(
      <OakModalNew
        title={"test_title"}
        content={"test_content"}
        footer={"test_footer"}
        open={false}
        onClose={onCloseMock}
      />,
    );

    const resetMocks = () => {
      closeMock.mockClear();
      showModalMock.mockClear();
      onCloseMock.mockClear();
    };

    expect(closeMock).not.toHaveBeenCalled();
    expect(showModalMock).not.toHaveBeenCalled();
    expect(onCloseMock).not.toHaveBeenCalled();

    resetMocks();
    rerender(
      <OakModalNew
        title={"test_title"}
        content={"test_content"}
        footer={"test_footer"}
        open={true}
        onClose={onCloseMock}
      />,
    );

    expect(closeMock).not.toHaveBeenCalled();
    expect(showModalMock).toHaveBeenCalled();
    expect(onCloseMock).not.toHaveBeenCalled();

    resetMocks();
    rerender(
      <OakModalNew
        title={"test_title"}
        content={"test_content"}
        footer={"test_footer"}
        open={false}
        onClose={onCloseMock}
      />,
    );

    expect(closeMock).toHaveBeenCalled();
    expect(showModalMock).not.toHaveBeenCalled();
    expect(onCloseMock).not.toHaveBeenCalled();
  });

  it("should open/close when open state changed", () => {
    const closeMock = jest.fn();
    const showModalMock = jest.fn();
    const onCloseMock = jest.fn();
    HTMLDialogElement.prototype.close = closeMock;
    HTMLDialogElement.prototype.showModal = showModalMock;
    const { container } = renderWithTheme(
      <OakModalNew
        title={"test_title"}
        content={"test_content"}
        footer={"test_footer"}
        open={false}
        onClose={onCloseMock}
      />,
    );

    act(() => {
      (container.querySelector('[aria-label="Close"]') as HTMLElement).click();
    });

    expect(onCloseMock).toHaveBeenCalled();
  });
});
