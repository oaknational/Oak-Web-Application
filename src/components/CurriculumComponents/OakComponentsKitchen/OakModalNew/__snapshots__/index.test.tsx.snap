// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakModalNew should match snapshot 1`] = `
<div>
  <dialog
    class="sc-692ca-0 jfQDcn"
    data-testid="modal"
    style="z-index: 320;"
  >
    <div
      class="sc-fqkvVR sc-f88aefea-0 hJRGVt hJveEH"
    />
    <div
      class="sc-fqkvVR sc-gsFSXq sc-2c6f78eb-0 hBVYLH hDihdg kDzojl"
      data-testid="sidebar-modal"
      direction="right"
    >
      <div
        class="sc-fqkvVR sc-gsFSXq kwqmrC ibSZJr"
      >
        <div
          class="sc-fqkvVR sc-gsFSXq kUmAOC dNHjLk"
        >
          <h3
            class="sc-dhKdcB kxDsEU"
            data-testid="modal-title"
          >
            test_title
          </h3>
          <div
            class="sc-fqkvVR RMQib"
          >
            <div
              class="sc-ad0ad6f5-0 sc-952e1041-0 bZaRlP hlZDaV"
            >
              <button
                aria-label="Close"
                class="sc-ce4d7bd4-0 sc-4362740-0 kYyTPx kgfbIy"
                data-testid="close-modal-button"
                title="Close"
              >
                <div
                  class="sc-b071a727-0 eQqUiM"
                >
                  <span
                    aria-hidden="true"
                    class="sc-cba1a45e-0 bgbQLJ"
                  >
                    <span
                      class="sc-cba1a45e-1 idsnjc"
                    >
                      <svg
                        aria-hidden="true"
                        class="sc-86d1d46f-0 eJtvEZ"
                        height="100%"
                        name="cross"
                        width="100%"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <use
                          xlink:href="/images/sprite/sprite.svg#cross"
                        />
                      </svg>
                    </span>
                  </span>
                  <svg
                    aria-hidden="true"
                    class="sc-86d1d46f-0 jOBmUC sc-62671a58-0 duckYG"
                    height="100%"
                    name="underline-1"
                    width="100%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <use
                      xlink:href="/images/sprite/sprite.svg#underline-1"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR sc-gsFSXq dWuEkM JfWBw"
          data-testid="modal-content"
        >
          test_content
        </div>
        <div
          class="sc-fqkvVR sc-gsFSXq eGXHla XmDSc"
          data-testid="modal-footer"
        >
          test_footer
        </div>
      </div>
    </div>
  </dialog>
</div>
`;
